### DISCUSSION TITLE
Multiple changes in one PR

### DISCUSSION THREAD

**Discussion by vedantmgoyal9** [microsoft/winget-pkgs#27411] [Ideas] [answered]
After reading https://github.com/microsoft/winget-pkgs/pull/26935#issuecomment-912833127, an idea came to my mind that if we can make a community moderator from https://github.com/microsoft/winget-pkgs/issues/15674#issue-908889039, a head moderator or give them additional permissions so that they can merge PRs like #26935 or PRs containing multiple small changes (like updating ARP or monikers). This will help to reduce the workload to approve multiple PRs for the same change. For example: when I raised issue #21256, Levvie had to open 400+ PRs and giving additional permissions to moderators will be helpful in these cases. 

Mentioning @ImJoakim @ItzLevvie @jedieaston @KaranKad @OfficialEsco @quhxl @denelon to give their opinions.

**Comment by KaranKad** [comment]
Improving validation process will also help with this issue. eg - 

- Not running VM to test each installer for existing manifest if changes are metadata-only so merging process can be fast.
- Allowing to make metdata-only changes to multiple maifests in one single PR.
- Allowing to delete multiple manifests in one single PR.

#18354

**Reply by vedantmgoyal9** [reply]
In addition to this, a contributor should be allowed to move or delete (not **add**) .validation files. A check should be done before that no version exists inside that Package Identifier.

**Reply by vedantmgoyal9** [reply]
@denelon can you put this in the backlog?

**Reply by denelon** [reply]
We're driving critical behavior with .validation files. They will continue to require one of the core team members to be removed/added.

**Comment by jedieaston** [comment]
ARP changes probably need to be tested, to make sure matching works. The rest of the metadata is probably safe to not test IMO. 

I think we've discussed having multiple manifests in one PR before (I don't have a citation for that though), but the problem is that it makes it too easy to sneak something in that could cause issues. The pipelines are set up to only test one manifest at a time right now, they'd probably have to be adjusted a lot. I think there's logic in having a break glass "oh crap we have to change the license for all instances of Docker Desktop" flow, but approvals should probably be restricted to people at MS. It shouldn't need to be used very often, instead we should continue finding ways to automate checking manifests when they are added (ARP stuff could be entirely automated, as an example).

**Reply by vedantmgoyal9** [reply]
Well, I think _multiple manifests_ is slightly different from _multiple changes_. When testing multiple manifests in one PR, the logic becomes too complex and relatively slow. But, as KaranKad said, we can allow changes like metadata and deleting multiple versions of packages in one PR.

**Reply by jedieaston** [reply]
> deleting multiple versions of packages in one PR.

Isn't that multiple manifests?

**Reply by vedantmgoyal9** [reply]
Deleting manifests doesn't need any tests like ARP and Manifest Content Validation as most of the time we are deleting them because they are not working properly. The InstallerUrl is 404 or the application does not install silently. What I think about multiple manifests is when we add/update packages in one PR.

**Comment by denelon** [comment, marked as answer]
We're still working through the implications of multiple manifests in a single PR. There are multiple edge cases. I've broken the publish pipeline at least once in the past when approving multiple deletes in one PR. It's still work in progress.

**Reply by Trenly** [reply]
Is there an issue which is tracking against this at all, or one which has the disposition of whether or not this will be completed? I personally would like to give an upvote to the idea, but I understand that there are many complexities underlying this. If this is work that will be completed eventually, I'd also like to help review the spec

**Reply by denelon** [reply]
https://github.com/microsoft/winget-pkgs/issues/125

**Reply by Trenly** [reply]
Thank you! I dont know why I couldn't find it earlier

**Comment by d3vel0per** [comment]
Good I saw this in discussions. I agree there should be a way out
eg. I had raised #28540 to change packaging of Temurin binaries due to the below reasons:

Vendor name change from `Eclipse Foundation` to `Eclipse Adoptium`
Product name change from `AdoptiumOpenJDK` to `Temurin`

Originally it was just one PR to move everything out from EclipseFoundation to EclipseAdoptium then based on CI pipeline failures:
1) Had to split into multiple PRs per version of JDK
2) And then again it's not allowing to move manifests across but delete and add them back in separate PRs

Eventually I closed it to pick up later.

That itself breaks the atomicity of the requirement. Wouldn't make much sense if one would look back at history as well
